### PR TITLE
multiple pages: fix French colon punctuation

### PR DESCRIPTION
**Pull request by bl-ue**
Ensures that there is always one space character before colons (`0xa0` thin unbreakable space instead of `0x20` normal space) in French pages.

### CODE REVIEW

File: pages.fr/common/7z.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 `7z a {{archive.7z}} {{chemin/vers/fichier_ou_dossier}}`
 
-- Chiffrer une archive existente (en incluant les en-têtes):
+- Chiffrer une archive existente (en incluant les en-têtes :
```
Comment (sbrl) [inline pages.fr/common/7z.md:10]: Typo?
```suggestion
- Chiffrer une archive existente (en incluant les en-têtes) :
```

File: pages.fr/common/hping.md [hunk 2]
```diff
@@ -4,22 +4,22 @@
 > Inspirer par la commande `ping`.
 > Plus d'informations : <http://www.hping.org>.
 
-- Ping localhost via TCP:
+- Ping localhost via TC :
```
Comment (sbrl) [inline pages.fr/common/hping.md:7]: I think your script went wrong.
```suggestion
- Ping localhost via TCP :
```

File: pages.fr/common/hping.md [hunk 3]
```diff
@@ -4,22 +4,22 @@
...
 
 `hping3 {{localhost}}`
 
-- Ping une adresse IP, via TCP, sur un port spécifique:
+- Ping une adresse IP, via TCP, sur un port spécifiqu :
```
Comment (sbrl) [inline pages.fr/common/hping.md:11]: I think your script went wrong.
```suggestion
- Ping une adresse IP, via TCP, sur un port spécifique :
```

File: pages.fr/common/hping.md [hunk 4]
```diff
@@ -4,22 +4,22 @@
...
 
 `hping3 -p {{80}} -S {{192.168.1.1}}`
 
-- Ping une adresse IP, via UDP, sur le port 80:
+- Ping une adresse IP, via UDP, sur le port 8 :
```
Comment (sbrl) [inline pages.fr/common/hping.md:15]:
```suggestion
- Ping une adresse IP, via UDP, sur le port 80 :
```

File: pages.fr/common/hping.md [hunk 5]
```diff
@@ -4,22 +4,22 @@
...
 
 `hping3 --udp -p {{80}} -S {{192.168.1.1}}`
 
-- Scanner un ensemble de ports TCP, sur une adresse IP spécifique:
+- Scanner un ensemble de ports TCP, sur une adresse IP spécifiqu :
```
Comment (sbrl) [inline pages.fr/common/hping.md:19]:
```suggestion
- Scanner un ensemble de ports TCP, sur une adresse IP spécifique :
```

File: pages.fr/common/hping.md [hunk 6]
```diff
@@ -4,22 +4,22 @@
...
 
 `hping3 --scan {{80,3000,9000}} -S {{192.168.1.1}}`
 
-- Effectuer un test de montée en charge sur le port 80:
+- Effectuer un test de montée en charge sur le port 8 :
```
Comment (sbrl) [inline pages.fr/common/hping.md:23]:
```suggestion
- Effectuer un test de montée en charge sur le port 80 :
```

File: pages.fr/common/nano.md [hunk 7]
```diff
@@ -3,22 +3,22 @@
 > Éditeur de texte simple et convivial. C'est un clone libre et amélioré de Pico.
 > Plus d'informations : <https://nano-editor.org>.
 
-- Ouvre un fichier:
+- Ouvre un fichie :
```
Comment (sbrl) [inline pages.fr/common/nano.md:6]:
```suggestion
- Ouvre un fichier :
```

File: pages.fr/common/nano.md [hunk 8]
```diff
@@ -3,22 +3,22 @@
...
 
 `nano {{chemin/vers/fichier}}`
 
-- Ouvre un fichier en positionnant le curseur à une rangée et colonne donnée:
+- Ouvre un fichier en positionnant le curseur à une rangée et colonne donné :
```
Comment (sbrl) [inline pages.fr/common/nano.md:10]:
```suggestion
- Ouvre un fichier en positionnant le curseur à une rangée et colonne donnée :
```

**Review by sbrl — DISMISSED** [review]
These are just a few of the issues - I think your regex went a bit wrong lol

The last character has been _replaced_ with a space, rather than a space being inserted.

Finally, does that strange space character mean anything in particular in French? Best to check before changing it.

**Comment by bl-ue** [comment]
@sbrl, you're right!! Funny 😄 

>Finally, does that strange space character mean anything in particular in French? Best to check before changing it.

I changed it because more pages have a normal space (139 pages) than ones that have the strange one (118), and that space was also in [an English page](https://github.com/tldr-pages/tldr/pull/5152/files#diff-32453cb35e57a1e6c386b6cad927f3da47c66b1b7d076879785db8d27a47043c)...

**Comment by bl-ue** [comment]
@sbrl I found a bunch more of that mistake so I just went back to the beginning and re-did it all.  Done now ;)

**Review by navarroaxel — APPROVED** [review]

**Review by schneiderl — APPROVED** [review]
I've once seen someone explaining this weird space here in tldr but I cannot remember what is the reason for it.

**Comment by bl-ue** [comment]
@schneiderl could you be thinking of https://github.com/tldr-pages/tldr/pull/3446#discussion_r336743337?

**Comment by bl-ue** [comment]
😬 I fear that I was wrong about the space...maybe there should be a `0xa0` / `160` char code space before every colon...cc @julienc91 @dgsb

**Comment by julienc91** [comment]
Indeed, as I said in the comment you reported, French typographic rules require this non-breaking space before some punctuation characters (":", ";", "?" and "!" in particular). Replacing those by a normal space (0x20) would potentially break the layout by putting the punctuation character alone at the beginning of a line.

**Comment by bl-ue** [comment]
@julienc91 okay, thank you very much.  I didn't know this.

@mebeim mentioned in https://github.com/tldr-pages/tldr/pull/3446#discussion_r338519017 "Indeed a space before and after the colon should be used" &mdash;so there should be a non-breaking space _after_ the colon as well as before?

**Comment by julienc91** [comment]
@mebeim A non-breaking space *before* the colon, and a normal space *after*.

**Comment by bl-ue** [comment]
Really nice, thank you @julienc91.

Is the space after the `:` colon _required_?  In the middle of a sentence obviously it would be, but what about at the end of one of our examples like this:

```
- Compresser un fichier ou un dossier :

`...`
```
We wouldn't need to add a space **after** that, would we?

**Comment by julienc91** [comment]
No space needed if you have a line return character right after. Regex style (with 0x0a and 0x20 being respectively the non breaking space and the normal space):

```
/[^:]+(0x0a):($|(0x20))/
```

**Comment by bl-ue** [comment]
Okay, thank you @julienc91.  This is very good to know.  Now I will proceed.

By the way, it's `0xa0 (160) ' '`, not `0x0a (10) '\n'` 😉

**Comment by bl-ue** [comment]
Done.

@julienc91 do these look right?
- https://github.com/tldr-pages/tldr/pull/5152/files#diff-1c61aa9a8397c4430c25df09cbaf6c7596bfacb1019fe0da6a93e5bca201dc92R15
- https://github.com/tldr-pages/tldr/pull/5152/files#diff-4cb665ce9eff7c93b1faf87f558b7e734162eb62ee852205f6c745792a545877R33
- https://github.com/tldr-pages/tldr/pull/5152/files#diff-7cde9138294422a6ce303d10c7f157ae8a460762e746999d85674f2d3a7b7249R23

**Comment by julienc91** [comment]
@bl-ue looks good to me! Also, "ex" is shorthand "example", while "ici" just means "here", so I guess it depends on the context? I didn't see any error in the 3 occurrences I found.

**Comment by bl-ue** [comment]
Okay! I was just wondering if perhaps someone who translated those pages missed something :)

Thank you very much @julienc91 for your help!  You've been extremely helpful!

**Review by navarroaxel — APPROVED** [review]

**Review by sbrl — APPROVED** [review]
Looks good to me! I assume the regular space was swapped for a non-breaking one right?

**Comment by bl-ue** [comment]
Exactly.

**Comment by bl-ue** [comment]
**Note:** Just in case anyone forgets, please **squash** this PR and **do not rebase** it.  The commit history is a little messy because I was learning the whole way through :)

**Pull request merged**